test_runner/regress/test_timeline_delete.py

```diff
@@ -440,9 +462,16 @@ def got_hangup_log_message():
     # ok, retry without failpoint, it should succeed
     ps_http.configure_failpoints((failpoint_name, "off"))
 
-    # this should succeed
-    ps_http.timeline_delete(env.initial_tenant, child_timeline_id, timeout=2)
-    # the second call will try to transition the timeline into Stopping state, but it's already in that state
-    env.pageserver.allowed_errors.append(
-        f".*{child_timeline_id}.*Ignoring new state, equal to the existing one: Stopping"
-    )
+    try:
+        ps_http.timeline_delete(env.initial_tenant, child_timeline_id)
+        env.pageserver.allowed_errors.append(
+            f".*{child_timeline_id}.*Ignoring new state, equal to the existing one: Stopping"
+        )
+    except PageserverApiException as e:
+        if e.status_code != 404:
+            raise e
+        else:
+            # mock_s3 was fast enough to delete before we got the request in
+            env.pageserver.allowed_errors.append(
+                f".*{child_timeline_id}.*Error processing HTTP request: NotFound: timeline not found"
+            )
```

Comment:
Hm, so, even with request coalescing, we get the expected timeout error in line 441.

But, with request coalescing, we still run the future to completion.

So, the original timeline delete from line 441 is still running at the time we set the failpoint to `off`.

Before your change, we knew (implicitly) that the original timeline delete was not running anymore.

So, I'd like to have that spelled out here, or, replace this entire `try/except` block with a poll for the initial timeline delete to finish (e.g., by polling timeline list + the local directory for the timeline being gone)

Reply:
Hmm, fair.

Reply:
allowed_error added in 70c27172

Reply:
I think the pattern to check for absence is to list the tenant's timelines, so, it always returns 200.

https://github.com/neondatabase/neon/blob/bfeb428d1b367351df11b7b17a75cff1c3c64ff6/test_runner/regress/test_neon_cli.py#L52-L57

Can even turn it into a one-liner

```
assert [env.initial_timeline]  = [timeline_id for (_, timeline_id) in env.neon_cli.list_timelines()]
```

Reply:
(Feel free to ignore, just wanted to let you know)

Reply:
Done in 9ab4e5ae